pkg/api/types.go

```diff
@@ -421,6 +425,19 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
```

Comment:
The collision case is that you create endpoints called "foo" then I create
a service called "foo" and the endpoints controller fails.
On Apr 8, 2015 6:24 AM, "Huamin Chen" <notifications@github.com> wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6174#discussion_r27968510>
> :
>
> > @@ -421,6 +425,19 @@ type NFSVolumeSource struct {
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> >  }
> >
> > +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
> > +type GlusterfsVolumeSource struct {
> > +	// Required: EndpointsName is the endpoint name that details Glusterfs topology
> > +	EndpointsName string `json:"endpoints"`
>
> the gluster cluster lies outside the kube cluster. what's the collision
> case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6174/files#r27968510>
> .
>